### PR TITLE
Update to 0.14.0

### PR DESCRIPTION
**Pull request by ChristopherBiscardi**
I started updating to the 0.14 rc release. Each commit has a link to a PR that enacted the change.

* [12997](https://github.com/bevyengine/bevy/pull/12997): rename `multi-threaded` feature to `multi_threaded`
* [12827](https://github.com/bevyengine/bevy/pull/12827) `RenderAssets<Image>` is now `RenderAssets<GpuImage>`
* [12732](https://github.com/bevyengine/bevy/pull/12732) `FloatOrd` is now in `bevy_math`
* [12889](https://github.com/bevyengine/bevy/pull/12889) Gpu Frustum Culling removed the `dynamic_offset` of `Transparent2d`
* [12453](https://github.com/bevyengine/bevy/pull/12453) Render Phases are now binned and sorted.
* [9202](https://github.com/bevyengine/bevy/pull/9202) SubApp access has changed from Result to Option
* [11698](https://github.com/bevyengine/bevy/pull/11698) `GpuImage` now uses UVec2s
* [12791](https://github.com/bevyengine/bevy/pull/12791) the primitive_topology field on GpuMesh was removed.
* [12216](https://github.com/bevyengine/bevy/pull/12216) introduced an argument `&mut MeshVertexBufferLayouts` to `get_mesh_vertex_buffer_layout`, which bevy_ecs_tilemap calls in `RenderChunk2d::prepare`
- [12163](https://github.com/bevyengine/bevy/pull/12163) bevy_color was created and Color handling has changed. Specifically `Color::as_linear_rgba_f32` has been removed and we have to go through `LinearRgb` using `.linear()`.
- [12582](https://github.com/bevyengine/bevy/pull/12582) divided `VisibleEntities` into separate lists. Specify the entity type we want
- [9202](https://github.com/bevyengine/bevy/pull/9202) changed world access to functions. [relevent line](https://github.com/bevyengine/bevy/pull/9202/files#diff-b2fba3a0c86e496085ce7f0e3f1de5960cb754c7d215ed0f087aa556e529f97fR640)
  - This also surfaced [12655](https://github.com/bevyengine/bevy/pull/12655) which removed `Into<AssetId<T>>` for `Handle<T>`. using a reference or .id() is the solution here.
 -  [12551](https://github.com/bevyengine/bevy/pull/12551) Removed `World::cell()`... but we don't need it or the replacement anyway.
- [12582](https://github.com/bevyengine/bevy/pull/12582) `check_visibility` must be implemented for the "renderable" tilemap entities
- [13289](https://github.com/bevyengine/bevy/pull/13489) introduced matrix naming changes for `view.view_proj`

## Current Status

Code compiles. Examples error out with https://github.com/bevyengine/bevy/issues/13728

<details><summary>old history</summary>

Once the fix in the above issue is added (the zstd + bevy_pbr features to the dev-dependencies bevy), the examples run but don't render

screenshot-2024-06-07-at-01 28 08@2x

</details>

Basic example is now rendering

screenshot-2024-06-07-at-11 27 38@2x

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "bevy_ecs_tilemap"
 description = "A tilemap rendering plugin for bevy which is more ECS friendly by having an entity per tile."
-version = "0.12.0"
```
Comment (ChristopherBiscardi) [inline Cargo.toml:4]: bevy_ecs_tilemap 13 never released to crates.io, so I moved this to 14 because I *think* versions track bevy versions?

**Comment by rparrett** [comment]
I think the current migration for:

> 12582 divided VisibleEntities into separate lists. Specify the entity type we want

Is part of the problem at the moment. I suspect we need to add our own visibility type and check system: `check_visibility::<WithTilemap>` or whatever.

But there may be something else going on too.

**Comment by ChristopherBiscardi** [comment]
@rparrett yeah, it was the check_visibility implementation that needed to be added. I checked a number of examples (but not all yet) and so far every one I've checked has rendered.
## basic

screenshot-2024-06-07-at-11 27 38@2x

## 3d_iso

screenshot-2024-06-07-at-11 30 10@2x

**Comment by ChristopherBiscardi** [comment]
I've updated color usage in examples as minimally as possible to make the tests and examples run, but I'd like to come back after this merges to update all of the example's colors to the new palettes and make some of the contrast more readable in a future PR if that's ok.

**Comment by rparrett** [comment]
I'll run through this checklist tonight:

- [ ] Check all examples on native (macos over here)
- [ ] Check all examples on native (with `features=atlas`)
- [ ] Spot check `basic`, `custom_shader`, `tiled`, `ldtk` with `--target=wasm32-unknown-unknown --features=atlas` 
- [ ] Spot check `basic`, `custom_shader`, `tiled`, `ldtk` with `--target=wasm32-unknown-unknown --features=bevy/webgpu`

**Comment by rparrett** [comment]
> I'd like to come back after this merges to update all of the example's colors to the new palettes and make some of the contrast more readable in a future PR if that's ok.

That would be nice. They have been in a particularly bad state since Bevy changed the default clear color.

**Comment by ChristopherBiscardi** [comment]
for windows, since you're doing mac later:

* [x]  Check all examples on native (macos over here) ([cargo-examples](https://crates.io/crates/cargo-examples) is nice)
* [x]  Check all examples on native (with `features=atlas`)
  - crashes on texture_container example, which seems to be the expected behavior as this combination is not supported
* [ ]  Spot check `basic`, `custom_shader`, `tiled`, `ldtk` with `--target=wasm32-unknown-unknown --features=atlas`

basic example has an issue in bevy_core_pipeline

```
error[E0432]: unresolved import `crate::core_3d::DEPTH_TEXTURE_SAMPLING_SUPPORTED`
  --> C:\Users\chris\.cargo\registry\src\index.crates.io-6f17d22bba15001f\bevy_core_pipeline-0.14.0-rc.2\src\dof\mod.rs:59:19
   |
59 |         Camera3d, DEPTH_TEXTURE_SAMPLING_SUPPORTED,
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `DEPTH_TEXTURE_SAMPLING_SUPPORTED` in `core_3d`
   |
note: found an item that was configured out
  --> C:\Users\chris\.cargo\registry\src\index.crates.io-6f17d22bba15001f\bevy_core_pipeline-0.14.0-rc.2\src\core_3d\mod.rs:53:11
   |
53 | pub const DEPTH_TEXTURE_SAMPLING_SUPPORTED: bool = false;
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: found an item that was configured out
  --> C:\Users\chris\.cargo\registry\src\index.crates.io-6f17d22bba15001f\bevy_core_pipeline-0.14.0-rc.2\src\core_3d\mod.rs:63:11
   |
63 | pub const DEPTH_TEXTURE_SAMPLING_SUPPORTED: bool = true;
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

* [x]  Spot check `basic`, `custom_shader`, `tiled`, `ldtk` with `--target=wasm32-unknown-unknown --features=bevy/webgpu`

all webgpu examples build. It doesn't look like we have a built-in setup to run them with cargo run so I did not run them.


---

also cargo-examples is amazing. very useful, thanks for the recommendation.

**Comment by ChristopherBiscardi** [comment]
wasm builds also require bevy/webgl2 because the const is also [gated by bevy_core_pipeline/webgl](https://github.com/bevyengine/bevy/blob/b17292f9d11cf3d3fb4a2fb3e3324fb80afd8c88/crates/bevy_core_pipeline/src/core_3d/mod.rs#L52). This was added in [13418](https://github.com/bevyengine/bevy/pull/13418)

```
〉cargo build --example custom_shader --target wasm32-unknown-unknown --features=atlas --features=bevy/webgl2
```

**Comment by rparrett** [comment]
Ah, that's why I suggested adding back `webgl2` to `dev-dependencies` above.

**Comment by ChristopherBiscardi** [comment]
oh you did? I must have missed that (and can't find it either)

**Comment by rparrett** [comment]
Oops, I had left my review pending.

**Comment by ChristopherBiscardi** [comment]
confirmed that fixed the build on my end after committing the review changes

**Comment by ChristopherBiscardi** [comment]
alright, I finished up the last of the clippy fixes too so this should be ready for you when you get to it later. Unless there's an issue I don't think there's anything left to do in this PR.

File: Cargo.toml (hunk 1 above)
Comment (ChristopherBiscardi) [reply Cargo.toml:4]: I can change the version to anything relevant. It seems like the Bevy project is hoping people will release rc candidates, although I don't know if anyone with publish access has the time for that at the moment for this crate.

File: Cargo.toml [hunk 2]
```diff
@@ -47,11 +47,14 @@ features = [
     "bevy_text",
     "bevy_sprite",
     #"file_watcher",
-    "multi-threaded",
+    "multi_threaded",
+    "webgl2",
+    "zstd",
```
Comment (rparrett) [inline Cargo.toml:52]:
```suggestion
    # TODO Remove these before release. See https://github.com/bevyengine/bevy/issues/13728
    "zstd",
```

Don't want this to get forgotten.

**Review by rparrett — APPROVED** [review]
Looks good to me! Awesome work.

File: Cargo.toml (hunk 1 above)
Comment (rparrett) [reply Cargo.toml:4]: I think StarArawn generally handles release prep himself.

**Comment by rparrett** [comment]
Opened https://github.com/ChristopherBiscardi/bevy_ecs_tilemap/pull/1 for a minor touchup removing the lint-allow from the `ldtk`/`tiled` examples.

**Comment by ChristopherBiscardi** [comment]
merged it in. Committed the removal of the second allow lint as well.

**Comment by ChristopherBiscardi** [comment]
- [ ] Color.linear changes to Color.to_linear when the next rc goes out: https://github.com/bevyengine/bevy/pull/13759

File: Cargo.toml [hunk 3]
```diff
@@ -47,11 +47,15 @@ features = [
     "bevy_text",
     "bevy_sprite",
     #"file_watcher",
-    "multi-threaded",
+    "multi_threaded",
+    "webgl2",
+    # TODO Remove these before release. See https://github.com/bevyengine/bevy/issues/13728
```
Comment (rparrett) [inline Cargo.toml:52]: Should be able to remove these now

**Pull request merged**